File: web_iot/python/dht11.py
# pip install pyserial flask
import serial
import flask
import json

app = flask.Flask(__name__, static_folder="../static")
dev_name = "/dev/tty.usbmodem14201"
speed = 9600
port = serial.Serial(dev_name, speed)
print("Serial device ready :", port.isOpen())

@app.route("/")
def index():
    return flask.redirect("/static/dht11.html")

@app.route("/monitor")
def monitor():
    port.write("m".encode())
    msg = port.readline().decode()
    ret = msg.split(",")
    ret = {"humi": ret[0], "temp":ret[1] }
    print(ret)
    return json.dumps(ret)

if __name__ == '__main__':
    app.run(host="0.0.0.0", port=8080)

File: web_iot/nodeJs/dht11.js
//npm install express serialport
const express = require('express');
const SerialPort = require("serialport");
const readline = require('readline');

let app = express();

var dev_name = "/dev/tty.usbmodem14201"
var speed = 9600;
let port_r;
var port = new SerialPort(dev_name, {baudRate:speed}, (err)=>{
    if(err) console.log("serial port open err", err);
    else port_r = readline.createInterface(port);
});


app.get('/', (req, res)=>{
    res.redirect('/dht11.html');
});
app.use(express.static(__dirname + '/../static/'));

app.get("/monitor", (req, res)=>{
    if(port.isOpen){
        port_r.once('line', (data)=>{
            console.log(data);
            ret = data.split(",")
            res.json({humi: ret[0], temp:ret[1]});
        });
        port.write('m');
    }
});

app.listen(8080, ()=>{
    console.log("server on 8080.");
});
